### PR TITLE
test on python 3.12

### PR DESCRIPTION
**Pull request by graingert**

### CODE REVIEW

**Comment by graingert** [comment]
so it turns out the 3.12 tests don't actually cover the build process

File: .github/workflows/ci.yaml [hunk 1]
```diff
@@ -54,10 +54,13 @@ jobs:
         python-version: ${{ matrix.python-version }}
         allow-prereleases: true
 
+    - name: Test build
+      run: pipx run --python=python build .
```
Comment (adiroiban) [inline .github/workflows/ci.yaml:58]: I see that `build` is called before the deps are installed.

Is that ok ?

**Review by adiroiban — APPROVED** [review]
Thanks. I think this is good to merge.

Only a minor comment.

Feel free to merge.

File: setup.py [hunk 2]
```diff
@@ -40,4 +44,5 @@ def read(*parts):
         maintainer='Twisted Matrix Labs Developers',
         version=versioneer.get_version(),
         cmdclass=versioneer.get_cmdclass(),
+        python_requires=">= 3.8",
```
Comment (adiroiban) [inline setup.py:47]: I t
I think that we also need to remove the universal build

The resulting wheel is

```
constantly-0+untagged.1.g6bbe661-py2.py3-none-any.whl
```

File: .github/workflows/ci.yaml (hunk 1 above)
Comment (graingert) [reply .github/workflows/ci.yaml:58]: pipx manages the deps

**Pull request merged**